frontend/src/features/codes/IndicesList.tsx

```diff
@@ -0,0 +1,258 @@
+            />
+            <div className={"index-actions"}>
+                <Button
+                    theme="black"
+                    iconLeft={<IconPlus />}
+                    onClick={() => setCreateDialogOpen(true)}
+                >
+                    Lisää/päivitä indeksi
+                </Button>
+            </div>
+            <EditIndexDialog
+                editDialogOpen={editDialogOpen}
+                closeDialog={closeDialog}
+                indexType={currentIndexType}
+                formData={formData}
+                setFormData={setFormData}
+            />
+        </div>
+    );
+};
+
+const EditIndexDialog = ({indexType, formData, setFormData, editDialogOpen, closeDialog}) => {
+    const [saveIndex, {data: saveData, error: saveError, isLoading: isSaving}] = useSaveIndexMutation();
+    const handleSaveIndex = () => {
+        saveIndex({
+            data: formData,
+            index: indexType.label,
+            month: formData.month,
+        });
+        closeDialog();
+    };
+    useEffect(() => {
+        if (isSaving) return;
+        if (saveData) {
+            hitasToast(
+                saveError ? "Indeksin tallennus epäonnistui" : "Indeksi tallennettu onnistuneesti",
+                saveError ? "error" : "success"
+            );
+        }
+    }, [isSaving, saveError, saveData]);
+    return isSaving ? (
+        <LoadingSpinner />
+    ) : (
+        <Dialog
+            id="index-creation-dialog"
+            aria-labelledby={"create-modal"}
+            isOpen={editDialogOpen}
+            close={() => closeDialog()}
+            closeButtonLabelText={"Sulje"}
+            boxShadow={true}
+        >
+            <Dialog.Header
+                id="index-creation-header"
+                title={"Tallenna " + getIndexTypeName(indexType.label).toLowerCase()}
+            />
+            <Dialog.Content>
+                <FormInputField
+                    label={"Kuukausi"}
+                    fieldPath={"month"}
+                    formData={formData}
+                    setFormData={setFormData}
+                    error={saveError}
+                    tooltipText={"Esim 2022-12"}
+                />
+                <FormInputField
+                    label={"Arvo"}
+                    inputType={"number"}
+                    fieldPath={"value"}
+                    formData={formData}
+                    setFormData={setFormData}
+                    error={saveError}
+                />
+            </Dialog.Content>
+            <Dialog.ActionButtons>
+                <Button
+                    onClick={() => closeDialog()}
+                    theme={"black"}
+                    variant={"secondary"}
+                >
+                    Peruuta
+                </Button>
+                <Button
+                    onClick={handleSaveIndex}
+                    theme={"black"}
+                >
+                    Tallenna
+                </Button>
```

Comment:
Or the loading indicator could alternatively be moved here with:

```
<SaveButton
    onClick={handleSaveIndex}
    isLoading={isSaving}
/>
```

Reply:
Implemented with requested component